### DISCUSSION TITLE
commands and date arguments

### DISCUSSION THREAD

**Discussion by jbaumgarten** [Rapptz/discord.py#9035] [Q&A] [answered]
Hi guys,

How date/time should be passed as arguments? Should we use strings or is there any class available?

**Comment by CaedenPH** [comment]
This is a tricky one because typically you would have to parse a human input string such as `2 days`, however if the user knows isoformat you could accept `float` type.
To my knowledge, there is no automatic converter.
However, you can use the `humanize` package on pypi to parse human string input

**Comment by DenverCoder1** [comment, marked as answer]
I believe you just have to use strings.

For slash commands, you can probably make a custom UI using a date parser (eg. [`dateparser`](https://github.com/scrapinghub/dateparser)) and auto-complete though.

**Reply by jbaumgarten** [reply]
> I believe you just have to use strings.
> 
> For slash commands, you can probably make a custom UI using a date parser (eg. [`dateparser`](https://github.com/scrapinghub/dateparser)) and auto-complete though.

I wanted to use basic slash commands but I'll try to use custom UI
